### PR TITLE
Fixity access modifier

### PR DESCRIPTION
**Pull request by andrevidela**
# Description

This is part of #2998 

- Allow export modifiers in front of a fixity declaration
- Update the parsing error message for "public" instead of "public export" 
- import order does not influence fixity parsing anymore

## Should this change go in the CHANGELOG?

- [x] If this is a fix, user-facing change, a compiler change, or a new paper
      implementation, I have updated `CHANGELOG.md` (and potentially also
      `CONTRIBUTORS.md`).

### CODE REVIEW

**Comment by buzden** [comment]
I can see it from the code, but maybe it should be documented somewhere explicitly, that by default fixity declaration is exported, as before.

**Comment by Z-snails** [comment]
As `public export` and `export` mean the same thing for fixity declarations, I don't think `public export` should be allowed. Not allowing it leads to a more consistent style, and there's no reason I can think of to prefer the longer way.
Alternatively this could be a warning (something like "`public` does not have any effect here, consider removing it")

**Comment by andrevidela** [comment]
@Z-snails I think your suggestion goes against the goal of #2998 which is to enable more functionality without introducing a new bespoke export/module system for fixities. It already is consistent with the fact that you can `public export` a definition with no implementation and that will effectively do the same as `export` since the computation will be stuck.

Adding a warning could be an interesting feature but I'm not sure it's strictly necessary. Until now, we've always used warnings  when we're unable to detect an error but we suspect there is one. The difference between `public export` and `export` isn't a source of error. Regardless of what we decide, we should probably discuss this on the #2998 issue itself.

**Comment by andrevidela** [comment]
@gallais all good to merge?

File: src/Idris/Resugar.idr [hunk 1]
```diff
@@ -47,7 +47,7 @@ mkOp tm@(PApp fc (PApp _ (PRef opFC kn) x) y)
        -- that the fixity context contains the name `(++)`
        let rootName = UN (Basic (nameRoot raw))
        let asOp = POp fc opFC kn (unbracketApp x) (unbracketApp y)
-       if not (null (lookupName rootName (infixes syn)))
+       if not (null (lookupName rootName (fixities syn)))
```
Comment (gallais) [inline src/Idris/Resugar.idr:50]: What if we have a (private prefix) (export definition) in one module and a similar
(export infix) (export definition) in another? When resugaring aren't we going to
resugar the prefix op as an infix one here? That would be really quite confusing.

Comment (andrevidela) [reply src/Idris/Resugar.idr:50]: Good point, let me see if I can create a test case for that

Comment (andrevidela) [reply src/Idris/Resugar.idr:50]: I can confirm prefix `!!` in `!! a` is desugared as `a !!`. In the previous version, it was desugared as `(!!) a`. I'll work on a fix

Comment (andrevidela) [reply src/Idris/Resugar.idr]: didn't have the time to look at this until now but using `infixes` instead of `fixities` solved it

**Pull request merged**